pygmt/tests/test_figure.py

```diff
@@ -3,22 +3,19 @@
 
 Doesn't include the plotting commands which have their own test files.
 """
+import importlib
 import os
 from pathlib import Path
 
-try:
-    import IPython
-except ImportError:
-    IPython = None  # pylint: disable=invalid-name
```

Comment:
Another option looking at https://stackoverflow.com/questions/61384752/how-to-type-hint-with-an-optional-import/77341843#77341843 seems to be something like:

```python
IPython: Any = None
try:
    import IPython
except ImportError:
    pass
```

Reply:
It's unclear what the recommended way is for importing an optional module. It seems sometimes pandas and xarray use the `has_xxx` bool variable:

https://github.com/pandas-dev/pandas/blob/feca574525b2c37fac8d1d26bf82d428b95630c8/pandas/compat/compressors.py#L11

https://github.com/pandas-dev/pandas/blob/feca574525b2c37fac8d1d26bf82d428b95630c8/pandas/io/formats/style.py#L84

https://github.com/pydata/xarray/blob/49bd63a8332c1930a866724a2968b2d880dae25e/xarray/core/ops.py#L15

but sometimes they also use the old way, which mypy should complain, e.g., 
https://github.com/pydata/xarray/blob/49bd63a8332c1930a866724a2968b2d880dae25e/xarray/coding/times.py#L31

Reply:
Looking at mypy (e.g. https://github.com/python/mypy/issues/1297#issuecomment-508593494), they also seem to recommend a try-except-else pattern?

Reply:
I don't think it's officially recommended and it's also more difficult to understand than `IPython = None` and `_HAS_IPYTHON = False`.

Reply:
Ok, let's go with `_HAS_IPython` unless there's a better recommended way :slightly_smiling_face:

That said, we should probably decide first if adding `mypy` is something we want (https://github.com/GenericMappingTools/pygmt/pull/2808), otherwise the changes here aren't really needed.

Reply:
Since we're going to add type hints (#2794, #2812), we definitely need a static type checker. There are many static type checkers, like [mypy](https://github.com/python/mypy), [pyright](https://github.com/microsoft/pyright）, and [pytype](https://github.com/google/pytype), in which mypy is the official one, so it's a good option to start.